delfin/drivers/ibm/storwize_svc/ssh_handler.py

```diff
@@ -0,0 +1,401 @@
+        try:
+            alert_model = dict()
+            alert_name = SSHHandler.handle_split(alert.get(
+                SSHHandler.OID_ERR_ID), ':', 1)
+            error_info = SSHHandler.handle_split(alert.get(
+                SSHHandler.OID_ERR_ID), ':', 0)
+            alert_id = SSHHandler.handle_split(error_info, '=', 1)
+            severity = SSHHandler.TRAP_SEVERITY_MAP.get(
+                alert.get(SSHHandler.OID_SEVERITY),
+                constants.Severity.INFORMATIONAL
+            )
+            alert_model['alert_id'] = str(alert_id)
+            alert_model['alert_name'] = alert_name
+            alert_model['severity'] = severity
+            alert_model['category'] = 'Fault'
+            alert_model['type'] = constants.EventType.EQUIPMENT_ALARM
+            alert_model['sequence_number'] = SSHHandler. \
+                handle_split(alert.get(SSHHandler.OID_SEQ_NUMBER), '=', 1)
+            timestamp = SSHHandler. \
+                handle_split(alert.get(SSHHandler.OID_LAST_TIME), '=', 1)
+            time_type = '%a %b %d %H:%M:%S %Y'
+            occur_time = int(time.mktime(time.strptime(
+                timestamp,
+                time_type)))
+            alert_model['occur_time'] = int(occur_time * SSHHandler.
+                                            SECONDS_TO_MS)
+            alert_model['description'] = alert_name
+            alert_model['resource_type'] = SSHHandler.handle_split(
+                alert.get(SSHHandler.OID_OBJ_TYPE), '=', 1)
+            alert_model['location'] = SSHHandler.handle_split(alert.get(
+                SSHHandler.OID_OBJ_NAME), '=', 1)
+            return alert_model
+        except Exception as e:
+            LOG.error(e)
+            msg = ("Failed to build alert model as some attributes missing "
+                   "in alert message:%s.") % (six.text_type(e))
+            raise exception.InvalidResults(msg)
+
+    def login(self):
+        try:
+            with self.ssh_pool.item() as ssh:
+                SSHHandler.do_exec('lssystem', ssh)
+        except Exception as e:
+            LOG.error("Failed to login ibm storwize_svc %s" %
+                      (six.text_type(e)))
+            raise e
+
+    @staticmethod
+    def do_exec(command_str, ssh):
+        """Execute command"""
+        try:
+            utils.check_ssh_injection(command_str)
+            if command_str is not None and ssh is not None:
+                stdin, stdout, stderr = ssh.exec_command(command_str)
+                res, err = stdout.read(), stderr.read()
+                re = res if res else err
+                result = re.decode()
+        except paramiko.AuthenticationException as ae:
+            LOG.error('doexec Authentication error:{}'.format(ae))
+            raise exception.InvalidUsernameOrPassword()
+        except Exception as e:
+            err = six.text_type(e)
+            LOG.error('doexec InvalidUsernameOrPassword error')
+            if 'timed out' in err:
```

Comment:
Why not keep a Map of error and exception. Will save multiple else if...

Reply:
sample:
class Excep():
    def foo(self):
        print("I am foo")
    def bar(self):
        print("I am bar")

ex = Excep()
mymap = {'foo':ex.foo(), 'bar':ex.bar()}
def test_s():
    X = 2
    try:
        if(X == 1):
            msg = 'foo'
            raise "foo"
        else:
            msg = 'bar'
            raise 'bar'
    except Exception as e:
        mymap[msg]


test_s()